### PR TITLE
Custom Unicycle Support (Beamlings)

### LINKED ISSUE #415: Beamlings Support
Beamlings lua support needs to be implemented. Using this issue to track that progress.

Please edit this to include what the requirements are.

### PR DESCRIPTION
**Pull request by OfficialLambdax**
Autosaves the custom beamling upon deletion of the beamling. Yet to be discussed if this is the desired way todo this. Also supports the other models. Whatever the player used last, becomes their default

### CODE REVIEW

**Comment by Starystars67** [comment]
I wonder could this assist with it for you?

https://github.com/BeamMP/BeamMP/blob/development/lua/ge/extensions/multiplayer/multiplayer.lua#L21

I notice that there is a `getPlayerUnicycle()` so maybe a custom implementation of this would be a good idea?

**Comment by Starystars67** [comment]
This PR needs the accompanying lua side of the functionality to complete this.

This is a requirement for the 4.10.0 update.

**Comment by Starystars67** [comment]
Sample of the options in the UI as default:
image

Please can others test it by loading up this PR in BeamNG and see if it lists the various unicycle config options for you under the multiplayer settings. 

@OfficialLambdax you may be best to test this?

**Comment by OfficialLambdax** [comment]
Introduces two Options within the Multiplayer Options.
1. A Selector for Unicycle configs (the user here can choose their default unicycle config)
> Please test. Upon choosing a new config, the entry will vanish and a small text will appear on the top right indicating that the new default has been set.
2. A Option to automatically save the latest used unicycle config upon despawning it.
> Please test. Is at default turned off. Will only work in a multiplayer server.

The default unicycle config is saved under `userdata/vehicles/unicycle/beammp_default.pc`. If no such file exists, beammp's own default is spawned. The default is a beamling config. Beamlings are not part of beammp. therefore it wont be able to spawn a beamling and will probably spawn the no mesh variant (didnt test myself).

**Comment by lionkor** [comment]
Please @ reviewers keep in mind a review includes testing (on a server)

**Comment by Starystars67** [comment]
I tested this just now, made a custom config and saved it as santa however I do not see it in the list, Am I missing something?
image

**Comment by OfficialLambdax** [comment]
Also made a custom config and named it Santa. Directly opened the multiplayer options after and got that entry
jhghjgjkghjghj

But given that you have a config "beammp_default" i suspect you using a outdated version. This entry should not show up https://github.com/BeamMP/BeamMP/pull/405/files#diff-b8637c4a303f4f23aa3d45b100fb95cd08945850627cf28e8de684f04059287bR28

File: lua/ge/extensions/MPConfig.lua [hunk 1]
```diff
@@ -17,6 +17,42 @@ local M = {}
 local Nickname = ""
 local PlayerServerID = -1
 
+local function getUnicycleConfigs()
```
Comment (Starystars67) [inline lua/ge/extensions/MPConfig.lua:23]: Please can comments / documentation be added to this function.
Keeps the codebase consistent.

File: lua/ge/extensions/MPConfig.lua [hunk 2]
```diff
@@ -17,6 +17,42 @@ local M = {}
...
+		if file ~= "beammp_default" then
+			tmp[file] = file..'.pc'
+		end
+	end
+
+	if tmp ~= settings.getValue('unicycleConfigs') then settings.setValue('unicycleConfigs', tmp) end -- multiplayer.partial ui will read this value
+	return tmp
+end
+
+local function setDefaultUnicycle(configFileName)
```
Comment (Starystars67) [inline lua/ge/extensions/MPConfig.lua:44]: Please can comments / documentation be added to this function.
Keeps the codebase consistent.

File: lua/ge/extensions/MPConfig.lua [hunk 3]
```diff
@@ -233,14 +274,26 @@ local function onDeserialized(data)
 	PlayerServerID = data.PlayerServerID
 end
 
+local function onSettingsChanged()
```
Comment (Starystars67) [inline lua/ge/extensions/MPConfig.lua:287]: Please can comments / documentation be added to this function.
Keeps the codebase consistent.

**Review by Starystars67 — CHANGES_REQUESTED** [review]
Tested in-game and working as expected.
Needs some commenting added but otherwise good to go!

**Review by Starystars67 — APPROVED** [review]

**Pull request merged**